internal/dataplane/parser/translate_httproute.go

```diff
@@ -210,8 +224,79 @@ func generateKongRoutesFromHTTPRouteRule(httproute *gatewayv1alpha2.HTTPRoute, r
 
 		// otherwise apply the hostnames to the route
 		r.Hosts = append(r.Hosts, hostnames...)
+		// attach the plugins to be applied to the given route
+		if len(plugins) != 0 {
+			if r.Plugins == nil {
+				r.Plugins = make([]kong.Plugin, len(plugins))
```

Comment:
Oh, that will allocate `n` (where `n=len(plugins)`) empty plugins in the slice. You want to preallocate the backing array:

```suggestion
				r.Plugins = make([]kong.Plugin, 0, len(plugins))
```

Reply:
oh It slipped out, I'll fix it